backend/schema/schema.go

```diff
@@ -57,6 +57,23 @@ func (s *Schema) ResolveDataRef(ref *DataRef) *Data {
 	return nil
 }
 
+func (s *Schema) ResolveDataRefMonomorphised(ref *DataRef) (*Data, error) {
+	data := s.ResolveDataRef(ref)
+	if data == nil {
+		return nil, fmt.Errorf("unknown data %v", ref)
+	}
+
+	if len(ref.TypeParameters) > 0 {
+		var err error
+		data, err = data.Monomorphise(ref.TypeParameters...)
```

Comment:
MBD but you could just do `return data.Monomorphise(ref.TypeParameters...)` here